feat: customise export dialog with UIOptions.canvasActions.export prop (#3658)
excalidraw/excalidraw: src/packages/excalidraw/CHANGELOG.md

## Excalidraw API

### Features

- Export dialog can be customised with [`UiOptions.canvasActions.export`](https://github.com/excalidraw/excalidraw/blob/master/src/packages/excalidraw/README.md#exportOpts) [#3658](https://github.com/excalidraw/excalidraw/pull/3658).

>   Also now [`UIOptions`](https://github.com/excalidraw/excalidraw/blob/master/src/packages/excalidraw/README.md#UIOptions) is memoized to avoid unnecessary rerenders.

Reviewer: ```suggestion
  Also, [`UIOptions`](https://github.com/excalidraw/excalidraw/blob/master/src/packages/excalidraw/README.md#UIOptions) is now memoized to avoid unnecessary rerenders.
```

#### BREAKING CHANGE

> - `UIOptions.canvasActions.saveAsScene` is now renamed to `UiOptions.canvasActions.export.saveFileToDisk`, by default this prop is true hence the **save file to disk** button is rendered inside export dialog.

Reviewer: ```suggestion
- `UIOptions.canvasActions.saveAsScene` was renamed to `UiOptions.canvasActions.export.saveFileToDisk`. Defaults to `true` hence the **save file to disk** button is rendered inside the export dialog.
```

Author: shouldn't it be "is" instead of "was" ?

Reviewer: it was done in the past. If you wanted to use present tense I think you'd have to use something like `A is now named B`.

> - `exportToBackend` is now renamed to `UIOptions.canvasActions.export.exportToBackend`. If this prop is not passed, the **shareable-link** button will not be rendered as before.

Reviewer: ```suggestion
- `exportToBackend` was renamed to `UIOptions.canvasActions.export.exportToBackend`. If this prop is not passed, the **shareable-link** button will not be rendered, same as before.
```
